lib/active_merchant/billing/gateways/vantiv_express.rb

```diff
@@ -0,0 +1,475 @@
+          xml.PaymentType PAYMENT_TYPE[options[:payment_type]] || options[:payment_type] if options[:payment_type]
+          xml.SubmissionType SUBMISSION_TYPE[options[:submission_type]] || options[:submission_type] if options[:submission_type]
+          xml.DuplicateCheckDisableFlag 1 if options[:duplicate_check_disable_flag].to_s == 'true' || options[:duplicate_override_flag].to_s == 'true'
+        end
+      end
+
+      def parse_eci(payment)
+        eci = payment.eci
+        eci[0] == '0' ? eci.sub!(/^0/, '') : eci
+      end
+
+      def market_code(money, options, network_token_eci)
+        return 3 if network_token_eci
+
+        MARKET_CODE[options[:market_code]] || options[:market_code] || 0
+      end
+
+      def add_lodging(xml, options)
+        if lodging = options[:lodging]
+          xml.ExtendedParameters do
+            xml.Lodging do
+              xml.LodgingAgreementNumber lodging[:agreement_number] if lodging[:agreement_number]
+              xml.LodgingCheckInDate lodging[:check_in_date] if lodging[:check_in_date]
+              xml.LodgingCheckOutDate lodging[:check_out_date] if lodging[:check_out_date]
+              xml.LodgingRoomAmount lodging[:room_amount] if lodging[:room_amount]
+              xml.LodgingRoomTax lodging[:room_tax] if lodging[:room_tax]
+              xml.LodgingNoShowIndicator lodging[:no_show_indicator] if lodging[:no_show_indicator]
+              xml.LodgingDuration lodging[:duration] if lodging[:duration]
+              xml.LodgingCustomerName lodging[:customer_name] if lodging[:customer_name]
+              xml.LodgingClientCode lodging[:client_code] if lodging[:client_code]
+              xml.LodgingExtraChargesDetail lodging[:extra_charges_detail] if lodging[:extra_charges_detail]
+              xml.LodgingExtraChargesAmounts lodging[:extra_charges_amounts] if lodging[:extra_charges_amounts]
+              xml.LodgingPrestigiousPropertyCode lodging[:prestigious_property_code] if lodging[:prestigious_property_code]
+              xml.LodgingSpecialProgramCode lodging[:special_program_code] if lodging[:special_program_code]
+              xml.LodgingChargeType lodging[:charge_type] if lodging[:charge_type]
+            end
+          end
+        end
+      end
+
+      def add_terminal(xml, options, network_token_eci = nil)
+        card_present = CARD_PRESENT_CODE[options[:card_present_code]] || options[:card_present_code] || 0
+
+        xml.Terminal do
+          xml.TerminalID options[:terminal_id] || '01'
+          xml.CardPresentCode network_token_eci ? 3 : card_present
+          xml.CardholderPresentCode network_token_eci ? 7 : 0
+          xml.CardInputCode network_token_eci ? 4 : 0
+          xml.CVVPresenceCode 0
+          xml.TerminalCapabilityCode network_token_eci ? 5 : 0
+          xml.TerminalEnvironmentCode network_token_eci ? 6 : 0
+          xml.MotoECICode network_token_eci || 7
+          xml.TerminalType 1 if network_token_eci
```

Comment:
Many of these seem to be based around `network_token_eci`, are all of these meant to be set differently only for network token? The name of some suggest they would be set by other values (ex: `CardholderPresentCode` I would expect to match a stored creds or cvv presence check and `MotoECICode` would match a `sca_exemption` check)

Reply:
Yes, I didn't notice this when I picked up the ticket. I'm updating this now based on their docs and the GSFs that we allow on this gateway.

Reply:
That does make sense, but I decided to just pull them from the GSFs to keep it in line with the current element integration. That deduction could be added in the future though, if desired.

Reply:
Alma mentioned that one big change from the existing integration to the new one is that the old ones used `String` enums but the new integration uses codes. Since you are pulling these values from the GSFs, does this mean the merchant needs to change logic to pass in `codes` instead of the `Strings`?

Reply:
amazing thank you 🙇🏽 this will be massive for moving people over seamlessly and hopefully deleting the old version

Reply:
Okie doke. This should be ready for another review now!